framework/utils/randomUtils.py

```diff
@@ -84,107 +86,156 @@ def testSampling(self, n=1e5):
   # this is needed for now since we need to split the stoch enviroments
   distStochEnv = findCrowModule('distribution1D').DistributionContainer.instance()
   boxMullerGen = BoxMullerGenerator()
+  # not sure if this boxMullerGen still needed, will be test this line after first commit
 
-def randomSeed(value,seedBoth=False):
+def randomSeed(value,seedBoth=False,engine=None):
   """
     Function to get a random seed
     @ In, value, float, the seed
+    @ In, engine, instance, optional, random number generator
     @ In, seedBoth, bool, optional, if True then seed both random environments
     @ Out, None
   """
-  if stochasticEnv == 'numpy':
-    global npStochEnv
-    npStochEnv = np.random.RandomState(value)
-  else:
-    crowStochEnv.seed(value)
-    # we must do the following now since there is no separation between the distribution (stoch eviroment) and the one here
-    ## TODO: split it in the sampler. Design needed.
-    distStochEnv.seedRandom(value)
+  replaceGlobalEnv=False
```

Comment:
A comment line about why we need `replaceGlobalEnv` before this line might help future developers understand how this is structured.

Reply:
I still kinda confused though.